cyclos_configuration/scripts/direct_debits/ExtensionPoint_TopupCreateDirectDebit.groovy

```diff
@@ -0,0 +1,29 @@
+import org.cyclos.entities.banking.Transaction
+import org.cyclos.entities.users.User
+import org.cyclos.model.banking.accounts.AccountOwner
+import org.cyclos.model.users.records.RecordDataParams
+import org.cyclos.model.users.records.UserRecordDTO
+import org.cyclos.model.users.recordtypes.RecordTypeVO
+import org.cyclos.model.users.users.UserLocatorVO
+
+/**
+ * Transaction extension point script to create a new directDebit user record whenever a 
+ * topup transaction is made from debiet to a user.
+ */
+
+AccountOwner user = binding.toOwner
+Transaction transaction = binding.transaction
+
+if ( ! user instanceof User ) {
+    // The toOwner should be a member. If it is not, something is wrong, so stop.
+    return
+}
+
+RecordDataParams params = new RecordDataParams(
+    user: new UserLocatorVO(id: user.id),
+    recordType: new RecordTypeVO(internalName: 'directDebit')
+)
+UserRecordDTO record = recordService.getDataForNew(params).dto
+def fields = scriptHelper.wrap(record)
+fields.transaction = transaction
+recordService.save(record)
```

Comment:
Is the Extension point on the topup transaction (Confirm Event) to create a new user record indeed only executed if the transaction is successfull?

Reply:
The extension point will be executed if there were no errors so far. In case of any errors, the transaction is rolled back.
However, it is not possible to be sure at the moment the extension point runs, if the transaction will be committed or not.
If you want to run some code only if the transaction is actually committed, you can use:

```groovy
scriptHelper.addOnCommit {
   // This block runs only AFTER committing the main transaction
   // The block itself runs outside any database transaction
}

// OR

scriptHelper.addOnCommitTransactional {
   // This block runs only AFTER committing the main transaction
   // The block itself runs on a new database transaction
}
```

Reply:
Thanks, Luis! I added `scriptHelper.addOnCommitTransactional{}`.